core/encoding.go

```diff
@@ -14,6 +14,16 @@ type Commitment struct {
 	*bn254.G1Point
 }
 
```

Comment:
I think we should do away with all of these custom types. My suggestion: 

We have 

```
type G1Point struct {
	*bn254.G1Point
}

type G2Point struct {
	*bn254.G2Point
}
```

And create seralization/deserialization methods for these. 

If we want to, we can also do some type aliasing, i.e. 

```
type LengthProof = G2Point
type RevealProof = G1Point

```

I think it's probably okay to fully drop the `Commitment` and `LengthCommitment` tpyes/aliases and just use the `G1Point` and `G2Point` types directly.

Reply:
I like this suggestion. 
What about creating a new type instead of type aliasing though (`type LengthProof G2Point`)? I think it's safer to use stronger types

Reply:
Doing the stronger typing here obviously has the disadvantage that you can't use any methods like serialization/deserialization of the `G2Point` directly on the `LengthProof` object. You'd have unwrap, call the method, or call the method and then wrap. 

I don't have a strong sense of the advantages of the stronger typing in this case, but I'm happy to defer to you.